util/systemd/systemd_linux.go

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) 2020 Tailscale Inc & AUTHORS All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// +build linux
+
+package systemd
+
+import (
+	"log"
+	"sync"
+
+	"github.com/mdlayher/sdnotify"
+)
+
+var getNotifyOnce struct {
+	sync.Once
+	v *sdnotify.Notifier
+}
+
+func notifier() *sdnotify.Notifier {
+	getNotifyOnce.Do(func() {
+		var err error
+		getNotifyOnce.v, err = sdnotify.New()
+		if err != nil {
+			log.Printf("systemd: systemd-notifier error: %v", err)
+		}
+	})
+	return getNotifyOnce.v
+}
+
+// Ready signals readiness to systemd. This will unblock service dependents from starting.
+func Ready() {
+	err := notifier().Notify(sdnotify.Ready)
```

Comment:
You could also provide another status message here alongside the readiness notification if it'd be helpful.